File: sdkwork-backend-sdk-swift/Sources/API/PptTemplateApi.swift
import Foundation

public class PptTemplateApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing PPT template
    public func update(body: PlusPptTemplateForm) async throws -> PlusApiResultPlusPptTemplateVO? {
        let response = try await client.put(ApiPaths.backendPath("/ppt/template"), body: body)
        return response as? PlusApiResultPlusPptTemplateVO
    }

    /// Create a new PPT template
    public func create(body: PlusPptTemplateForm) async throws -> PlusApiResultPlusPptTemplateVO? {
        let response = try await client.post(ApiPaths.backendPath("/ppt/template"), body: body)
        return response as? PlusApiResultPlusPptTemplateVO
    }

    /// Get PPT templates by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusPptTemplateVO? {
        let response = try await client.post(ApiPaths.backendPath("/ppt/template/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusPptTemplateVO
    }

    /// Get all PPT templates
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusPptTemplateVO? {
        let response = try await client.post(ApiPaths.backendPath("/ppt/template/list/all"), body: body)
        return response as? PlusApiResultListPlusPptTemplateVO
    }

    /// Get a PPT template by ID
    public func getById(id: String) async throws -> PlusApiResultPlusPptTemplateVO? {
        let response = try await client.get(ApiPaths.backendPath("/ppt/template/\(id)"))
        return response as? PlusApiResultPlusPptTemplateVO
    }

    /// Delete a PPT template
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/ppt/template/\(id)"))
        return response as? PlusApiResultBoolean
    }
}

File: sdkwork-backend-sdk-csharp/Api/PptTemplateApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class PptTemplateApi
    {
        private readonly HttpClient _client;

        public PptTemplateApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing PPT template
        /// </summary>
        public async Task<PlusApiResultPlusPptTemplateVO?> UpdateAsync(PlusPptTemplateForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusPptTemplateVO>(ApiPaths.BackendPath("/ppt/template"), body);
        }

        /// <summary>
        /// Create a new PPT template
        /// </summary>
        public async Task<PlusApiResultPlusPptTemplateVO?> CreateAsync(PlusPptTemplateForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusPptTemplateVO>(ApiPaths.BackendPath("/ppt/template"), body);
        }

        /// <summary>
        /// Get PPT templates by page
        /// </summary>
        public async Task<PlusApiResultPagePlusPptTemplateVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusPptTemplateVO>(ApiPaths.BackendPath("/ppt/template/list"), body, query);
        }

        /// <summary>
        /// Get all PPT templates
        /// </summary>
        public async Task<PlusApiResultListPlusPptTemplateVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusPptTemplateVO>(ApiPaths.BackendPath("/ppt/template/list/all"), body);
        }

        /// <summary>
        /// Get a PPT template by ID
        /// </summary>
        public async Task<PlusApiResultPlusPptTemplateVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusPptTemplateVO>(ApiPaths.BackendPath($"/ppt/template/{id}"));
        }

        /// <summary>
        /// Delete a PPT template
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/ppt/template/{id}"));
        }
    }
}
